daemon/daemon.go

```diff
@@ -20,13 +21,19 @@ type Daemon struct {
 	server *http.Server
 	l      net.Listener
 	doneCh chan struct{}
+	e      *engine.Engine
 }
 
 func New(cfg *config.EnvConfig) (srv *Daemon, err error) {
 	srv = new(Daemon)
 
 	r := mux.NewRouter().StrictSlash(true)
 
+	srv.e, err = engine.New(cfg)
```

Comment:
Do we need Daemon? It appears to be creating an engine and starting an HTTP server which doesn't currently serve anything. I don't believe we should be starting a server on the client. Unless there is a reason we need a server on both sides, we could just construct the engine from the daemon command.

Reply:
We are starting an HTTP server on the `dealbot daemon` just so that we serve metrics in the future, and currently we block on the HTTP server listen I believe. You are right, that it doesn't do anything at the moment, but I think it will come handy very soon - for metrics, health, etc.

Reply:
The metrics could be pushed to the controller. We should keep our communication patterns standard, otherwise we're just going to complicate deployment.